### ISSUE TITLE
Readonly input fields should be focusable

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#3816]
Following #3583, it is possible to tab through `xf:output` fields that have a label, but readonly input fields do not support tabs!

One reason is that we are using `disabled` instead of `readonly` on readonly fields. They should be `readonly`, in which case they would become tabbable.

- [x] use `readonly`
    - [x] `input` and `textarea`
    - [x] what about other fields like `select` (not per spec)?
- [x] server must support focus on readonly field, as it does for `xf:output` now

**Comment by ebruchez** [comment]
- Client: We use `ORBEON.xforms.Controls.setDisabledOnFormElement` to set the `disabled` attribute. All callers do this from `setReadonly`. As a first step, try switching to `readonly`.
- Server: This is done from `outputDisabledAttribute`. The pattern is always:
    ```scala
    if (isHTMLDisabled(control))
      outputDisabledAttribute(containerAttributes)
    ```

[MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly):

> text, search, url, tel, email, password, date, month, week, time, datetime-local, and number \<input> types and the <textarea>

> Only text controls can be made read-only, since for other controls (such as checkboxes and buttons) there is no useful distinction between being read-only and being disabled, so the readonly attribute does not apply.

**Comment by ebruchez** [comment]
Confirming that the use case of #4481 works when implementing this.

Now, creating a sample with many field types disabled, other controls don't support tabbing:

- [ ] Number
    - [x] client-side
    - [ ] server-side: client dispatches focus event to XBL control, so it's rejected
- [ ] Date
- [ ] Dropdown
- [ ] Radio / Checkbox
- [ ] File
- [ ] Button

**Comment by ebruchez** [comment]
With Chrome:

- `<select>` with `readonly`: shows the dropdown when clicked, but you can't select a value
- `<input type="radio">` with `readonly`: allows selecting the radio button
- `<input type="checkbox">`: same

The HTML spec doesn't document that `readonly` should do anything on those controls, but I wanted to check whether they are doing something useful. Arguably, they are not (especially for radio/checkbox), so, for now, those are simply not supported.

This means that this will only fix #4481 for a limited number of controls.
